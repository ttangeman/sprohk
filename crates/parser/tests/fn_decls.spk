fn foo();
fn bar() -> i32;
fn x(a: i32, b: f32) -> f32;
fn y(a: i32) {}