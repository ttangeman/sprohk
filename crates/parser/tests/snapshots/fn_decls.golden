node 0 {
  kind: FnPrototype
  tokens: [0..4]
  data: {
    name: foo
  }
}
node 1 {
  kind: Function
  tokens: [0..4]
  data: {
    prototype: 0
  }
}
node 2 {
  kind: TypeExpr
  tokens: [10..11]
  data: {
    root: 'i32'
  }
}
node 3 {
  kind: FnPrototype
  tokens: [5..11]
  data: {
    name: bar
    ret_type_expr: 2
  }
}
node 4 {
  kind: Function
  tokens: [5..11]
  data: {
    prototype: 3
  }
}
node 5 {
  kind: TypeExpr
  tokens: [17..18]
  data: {
    root: 'i32'
  }
}
node 6 {
  kind: FnParameter
  tokens: [15..18]
  data: {
    name: a
    type_expr: 5
  }
}
node 7 {
  kind: TypeExpr
  tokens: [21..22]
  data: {
    root: 'f32'
  }
}
node 8 {
  kind: FnParameter
  tokens: [19..22]
  data: {
    name: b
    type_expr: 7
  }
}
node 9 {
  kind: TypeExpr
  tokens: [24..25]
  data: {
    root: 'f32'
  }
}
node 10 {
  kind: FnPrototype
  tokens: [12..25]
  data: {
    name: x
    ret_type_expr: 9
    params: [
      6
      8
    ]
  }
}
node 11 {
  kind: Function
  tokens: [12..25]
  data: {
    prototype: 10
  }
}
node 12 {
  kind: TypeExpr
  tokens: [31..32]
  data: {
    root: 'i32'
  }
}
node 13 {
  kind: FnParameter
  tokens: [29..32]
  data: {
    name: a
    type_expr: 12
  }
}
node 14 {
  kind: FnPrototype
  tokens: [26..33]
  data: {
    name: y
    params: [
      13
    ]
  }
}
node 15 {
  kind: Block
  tokens: [33..35]
  data: {
    statements: [
    ]
  }
}
node 16 {
  kind: Function
  tokens: [26..35]
  data: {
    prototype: 14
    block: 15
  }
}
